samples/bluetooth/peripheral_hr/src/main.c

```diff
@@ -56,7 +56,7 @@ static void connected(struct bt_conn *conn, uint8_t err)
 
 static void disconnected(struct bt_conn *conn, uint8_t reason)
 {
-	printk("Disconnected (reason 0x%02x)\n", reason);
+	printk("Disconnected (reason " HCI_ERR_FMT ")\n", HCI_ERR(reason));
```

Comment:
IMO, this looks a bit ugly, so I'm open for other suggestions!

Reply:
I have been trough a similar though process and decided to just print both. In the case of `CONFIG_BT_HCI_ERR_AS_STRING`, the function can be macroed out to the empty string, which I hope optimizes nicely.

https://github.com/zephyrproject-rtos/zephyr/blob/786b9a0ad405c3475382545aa29e3b815129e68a/subsys/bluetooth/shell/gatt.c#L1013-L1014

Reply:
Good suggestion. I like that better than mine. It also makes it more clear that it that it is possible to enable and disable string representations

Reply:
I would have hoped the compiler could have optimized the following a bit better:
```c
int main(void) {
    printf("Hello world %s\n", "");
}
```
Godbolt shows that using `ARM GCC 14.1.0` with options `-O3 -Os -Wall`, results in:
```asm
.LC0:
        .ascii  "\000"
.LC1:
        .ascii  "Hello world %s\012\000"
main:
        push    {r3, lr}
        ldr     r1, .L2
        ldr     r0, .L2+4
        bl      printf
        movs    r0, #0
        pop     {r3, pc}
.L2:
        .word   .LC0
        .word   .LC1
```.
That is, the empty string is not optimized away